spark/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/Functions.scala

```diff
@@ -1437,6 +1439,60 @@ case class ST_ForceRHR(inputExpressions: Seq[Expression])
   }
 }
 
+case class ST_GeneratePoints(inputExpressions: Seq[Expression], randomSeed: Option[Long] = None)
+    extends Expression
+    with Nondeterministic
+    with ExpectsInputTypes
+    with CodegenFallback
+    with ExpressionWithRandomSeed {
+
+  def this(inputExpressions: Seq[Expression]) = this(inputExpressions, Some(0L))
+
+  def seedExpression: Expression = randomSeed.map(Literal.apply).getOrElse(Literal(0L))
+
+  @transient private[this] var random: Random = _
+
+  protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]) = {
+    copy(inputExpressions = newChildren)
+  }
+
+  private val nArgs = children.length
+
+  override protected def initializeInternal(partitionIndex: Int): Unit = random = new Random(
+    randomSeed.getOrElse(0L) + partitionIndex)
+
+  override protected def evalInternal(input: InternalRow): Any = {
+    val geom = children.head.toGeometry(input)
+    val numPoints = children(1).eval(input).asInstanceOf[Int]
+    if (nArgs == 3) {
+      val seed = children(2).eval(input).asInstanceOf[Int]
+      return GeometrySerializer.serialize(Functions.generatePoints(geom, numPoints, seed))
+    }
+    GeometrySerializer.serialize(Functions.generatePoints(geom, numPoints))
+  }
```

Comment:
We should construct an instance of `RandomPointsBuilderSeed` in the `initializeInternal` function, and call    `setExtent`, `setNumPoints`, `getGeometry` methods using that instance in `evalInternal`.